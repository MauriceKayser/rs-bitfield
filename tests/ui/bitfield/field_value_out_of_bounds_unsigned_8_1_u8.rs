//! Assertion failure: _TYPE_IN_FIELD_0_EXCEEDS_FIELD_SIZE_OF_1_BIT

extern crate alloc;

#[bitfield::bitfield(8)]
struct BitField(#[field(size = 1)] Field); // Can only store values between `0..=1`.

#[derive(Clone, Copy, Debug, bitfield::Field)]
#[repr(u8)]
enum Field {
    F2 = 2
}

fn main() {}