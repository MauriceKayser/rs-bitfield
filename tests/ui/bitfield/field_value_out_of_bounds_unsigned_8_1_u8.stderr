error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/field_value_out_of_bounds_unsigned_8_1_u8.rs:6:36
  |
6 | struct BitField(#[field(size = 1)] Field); // Can only store values between `0..=1`.
  |                                    ^^^^^ the evaluated program panicked at 'Size of the field type exceeds the specified field size of 1 bit', $DIR/tests/ui/bitfield/field_value_out_of_bounds_unsigned_8_1_u8.rs:6:36
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
